tests/unit/domain/test_message_operations_service_part1.py

```
@@ -0,0 +1,374 @@
+from src.myjarvis.domain.value_objects.message_collection import (
+    MessageCollection as MC,
+)
+from src.myjarvis.domain.value_objects.message_role import MessageRole
+
+
+class FakeLimitsService:
+    def apply_limits(self, messages, limits):
+        if limits is None:
+            return messages
+        if limits.max_messages is not None:
+            messages = messages[-limits.max_messages :]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)